### PR TITLE
Keating build upgrade

### PR DESCRIPTION
**Pull request by brossetti1**
## GitHub Issue

https://github.com/HausDAO/daohaus-monorepo/issues/969

## Changes

- [X] create the executor for the custom packageJson update
- [X] include changes for vite builds
- [X] cleanup packageJsons as seen in @keatings `generate-deps` pr

The executor is stored in `tools/executors` - here is a better idea of where the executor plugs in:

Screen Shot 2022-10-12 at 12 17 28 PM
Screen Shot 2022-10-12 at 12 18 17 PM
Screen Shot 2022-10-12 at 12 20 04 PM

## Packages Added

None

## Checks

Before making your PR, please check the following:

- [ ] Critical lint errors are resolved
- [ ] App runs locally
- [ ] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

**Comment by brossetti1** [comment]
@alexkeating can you check this work

basically pulled alot of it from your branches from a while back:

https://github.com/HausDAO/daohaus-monorepo/compare/develop...feature/generate-deps
https://github.com/nxext/nx-extensions/pull/801/files

File: libs/common-utilities/project.json [hunk 1]
```diff
@@ -16,6 +16,14 @@
...
       }
     },
+    "vite-build": {
+      "dependsOn": [
+        {
+          "target": "build",
+          "projects": "self"
+        }
+      ]
+    },
```
Comment (brossetti1) [inline libs/common-utilities/project.json:26]: not sure if this is needed

File: libs/dao-data/project.json [hunk 2]
```diff
@@ -22,6 +22,14 @@
...
       }
     },
+    "vite-build": {
+      "dependsOn": [
+        {
+          "target": "build",
+          "projects": "self"
+        }
+      ]
+    },
```
Comment (brossetti1) [inline libs/dao-data/project.json:32]: not sure if this is needed - dao-data does not use vite

File: libs/daohaus-connect-feature/project.json [hunk 3]
```diff
@@ -5,14 +5,17 @@
...
       "options": {
         "outputPath": "dist/libs/daohaus-connect-feature",
         "frameworkConfigFile": "@nxext/react/plugins/vite",
         "configFile": "libs/daohaus-connect-feature/vite.config.ts",
         "entryFile": "src/index.ts",
-        "generatePackageJson": true
+        "updateBuildableProjectDepsInPackageJson": true,
+        "buildableProjectDepsInPackageJsonType": "dependencies",
+        "packageJson": "libs/daohaus-connect-feature/package.json",
+        "project": "libs/daohaus-connect-feature"
```
Comment (brossetti1) [inline libs/daohaus-connect-feature/project.json:18]: here is the change where we add our exectutor from `tools/executor` --- is this naming ok? 

let me know if you think of a better one

File: libs/dao-data/project.json (hunk 2 above)
Comment (skuhlmann) [reply libs/dao-data/project.json:32]: yeah that doesn't seem necessary

**Review by skuhlmann — COMMENTED** [review]
looking good - we can test with an external app soon.

File: libs/dao-data/project.json (hunk 2 above)
Comment (brossetti1) [reply libs/dao-data/project.json:32]: hmm weird, looks like we do need this?

got to be something we can do about it -- ill look into it

Screen Shot 2022-10-12 at 5 02 22 PM

Comment (alexkeating) [reply libs/dao-data/project.json:32]: That error basically means that command needs to exist for libraries that it depends on. That's why I added it here

File: libs/daohaus-connect-feature/project.json (hunk 3 above)
Comment (alexkeating) [reply libs/daohaus-connect-feature/project.json:18]: Might be better to name it @daohaus/nx-plugin as I think the package could contain both executors and generators

**Review by alexkeating — COMMENTED** [review]
Looks good to me

**Review by skuhlmann — APPROVED** [review]

**Pull request merged**